pkg/rtc/uptrackmanager.go

```diff
@@ -432,6 +429,11 @@ func (u *UptrackManager) handleTrackPublished(track types.PublishedTrack) {
 		delete(u.publishedTracks, trackSid)
 		delete(u.pendingSubscriptions, trackSid)
 		// not modifying subscription permissions, will get reset on next update from participant
+
+		// as rtcpCh handles RTCP for all published tracks, close only after all published tracks are closed
+		if len(u.publishedTracks) == 0 {
+			close(u.rtcpCh)
```

Comment:
@ashellunts mentioned in a different context about not getting TrackUnpublished event some times. Wondering if there are situations (like publisher network black out) where the SRTP session is not closed. That would mean OnClose not firing for the up track and this would not happen. Is that a possibility? How should we recover? Maybe force close up tracks in a clean up worker?

Reply:
this doesn't seem to be the right place to close it. A participant may still maintain the RTCP connection even if they have unpublished all of their tracks.

with the current design, the RTCP channel is shared across all published tracks. 

It seems unlikely that this callback would be missed. We've profiled this previously and have not seen leaks here.

Reply:
My bad. Will add an extra check.

Reply:
Adding another check in this commit - https://github.com/livekit/livekit-server/pull/286/commits/f8e61f11f614d561bc6f3e2b9fa9b08f662ead2b